src/mame/includes/jpmsys5.h

```diff
@@ -61,30 +77,67 @@ class jpmsys5_state : public driver_device
 	optional_device<s16lf01_device> m_vfd;
 	required_device<upd7759_device> m_upd7759;
 
+	optional_ioport m_dsw;
+	optional_ioport m_dsw2;
+	optional_ioport m_rotary;
+	optional_ioport m_strobe0;
+	optional_ioport m_strobe1;
+	optional_ioport m_strobe2;
+	optional_ioport m_strobe3;
+	optional_ioport m_strobe4;
```

Comment:
Is there a specific reason these aren’t an `optional_ioport_array`?

Reply:
i'm unsure what the final form will be here, this is just how they're named in the input layouts for other emulators, but really even the ports before this might be part of the same thing,   I'll probably commit to an array at some point, but it feels premature